CMakeLists.txt

```
@@ -16,6 +16,47 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rosidl_default_generators REQUIRED)
 
+set(protobuf_MODULE_COMPATIBLE TRUE)
+find_package(Protobuf CONFIG REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
+
+set(_PROTOBUF_LIBPROTOBUF protobuf::libprotobuf)
+set(_REFLECTION gRPC::grpc++_reflection)
+set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
+
+find_package(gRPC CONFIG REQUIRED)
+message(STATUS "Using gRPC ${gRPC_VERSION}")
+
+set(_GRPC_GRPCPP gRPC::grpc++)
+set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
+
+# Proto file
+get_filename_component(aruku_proto "./proto/aruku.proto" ABSOLUTE)
+get_filename_component(aruku_proto_path "${aruku_proto}" PATH)
+
+# Generated sources
+add_custom_command(
+      OUTPUT "${${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.cc}" "${${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.h}" "${${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.h}" "${${CMAKE_CURRENT_BINARY_DIR}/aruku.grpc.pb.h}"
+      COMMAND ${_PROTOBUF_PROTOC}
+      ARGS --grpc_out "${CMAKE_CURRENT_BINARY_DIR}"
+        --cpp_out "${CMAKE_CURRENT_BINARY_DIR}"
+        -I "${aruku_proto_path}"
+        --plugin=protoc-gen-grpc="${_GRPC_CPP_PLUGIN_EXECUTABLE}"
+        "${aruku_proto}"
+      DEPENDS "${aruku_proto}")
+
+add_library(aruku_proto
+  ${${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.cc}
+  ${${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.h}
+  ${${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.h}
+  ${${CMAKE_CURRENT_BINARY_DIR}/aruku.grpc.pb.h})
```

Why there's double `${}` in here? Won't it work?

```suggestion
  ${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.cc
  ${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.h
  ${CMAKE_CURRENT_BINARY_DIR}/aruku.pb.h
  ${CMAKE_CURRENT_BINARY_DIR}/aruku.grpc.pb.h)
```